portfolio/src/main/webapp/script.js

```diff
@@ -35,34 +35,34 @@ function addRandomGreeting() {
 
 /**
  * Loads the HTML retrieved from `src` url and inserts it at the `position` relative to `selector`
- * Example: loadHTML("templates/header.html", "body", "beforebegin") inserts the header before <body>
+ * Example: loadHTML("templates/header.html", "body", Adjacent.BEFORE);
+ * inserts the header before <body>
  */
 function loadHTML(src, selector, position) {
-  fetch(src)
+  return fetch(src)
     .then(response => response.text())
     .then(text => {
-      document.querySelector(selector).insertAdjacentHTML(position, text);
+      document.querySelector(selector)?.insertAdjacentHTML(position, text);
     });
 }
-
 // "Enum" for key positions used in insertAdjacentHTML() and loadHTML()
 const Adjacent = {
-  BEFORE_BEGIN: "beforebegin",
-  AFTER_BEGIN: "afterbegin",
-  BEFORE_END: "beforeend",
-  AFTER_END: "afterend"
+  BEFORE: "beforebegin",
+  PREPEND: "afterbegin",
+  APPEND: "beforeend",
+  AFTER: "afterend"
 };
  
-/**
- * Fetches and loads header, footer and other dynamic content for each page.
- */
+/** Fetches and loads header, footer and other dynamic content for each page. */
 function onBodyLoad() {
-  // Fetch & load header
-  loadHTML("templates/header.html", "body", Adjacent.BEFORE_BEGIN);
-  // Fetch & load footer
-  loadHTML("templates/footer.html", "body", Adjacent.AFTER_END);
-  // Fetch & load comments from /data
-  loadComments();
+  // Fetch & load header template
+  loadHTML("templates/header.html", "body", Adjacent.BEFORE)
+  // Fetch & load comments template and data from /data for authenticated users
+  // Note: A feature depends on the header being loaded, so it is necessary to await
+  // for the header fetch to resolve.
```

Comment:
This is actually for a new issue 😆  it showed up when I tried to include a Login/Logout button in the header.
The image upload problem seems to be a bigger issue. Sometimes fetch() requests will just get 
[stuck on pending ](https://user-images.githubusercontent.com/44207690/84394886-82d61380-abcb-11ea-83f8-e323422c181c.png) and never resolve. I'm going to test if the same issue occurs on a deployed server, tomorrow. If it continues, I have a feeling it might have to do with not `await`ing the fetches, which means I'd have to make my functions ["Async all the way up"](https://stackoverflow.com/questions/29808915/why-use-async-await-all-the-way-down)

Reply:
I found this answer https://stackoverflow.com/a/21252544
It looks like It's a problem with caching in Chrome .
I've disabled caching and it does not seem to be happening anymore.